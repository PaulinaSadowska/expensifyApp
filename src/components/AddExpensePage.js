import React from 'react';

export const AddExpensePage = () => (
    <p>add expense page</p>
);