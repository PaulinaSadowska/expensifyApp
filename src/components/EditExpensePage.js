import React from 'react';

export const EditExpensePage = (props) => {
    console.log(props)
    return <div>
        Editing the expense with id = {props.match.params.id}
        <button onClick={props.history.goBack}>back</button>
    </div>
};