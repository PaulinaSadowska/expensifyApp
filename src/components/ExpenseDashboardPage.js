import React from 'react';

export const ExpenseDashboardPage = () => (
    <p>expense dashboard page</p>
);