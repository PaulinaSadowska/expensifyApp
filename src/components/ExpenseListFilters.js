import React from 'react';
import { DateRangePicker } from 'react-dates';
import { connect } from 'react-redux'
import { setTextFilter, sortByDate, sortByAmount, setStartDate, setEndDate } from '../actions/filters'

export class ExpenseListFilters extends React.Component {

    state = {
        calendarFocused: undefined
    }

    onFocusChange = (focused) => {
        this.setState(() => ({
            calendarFocused: focused
        }))
    }

    onDatesChanged = ({ startDate, endDate }) => {
        console.log(startDate)
        this.props.setStartDate(startDate)
        this.props.setEndDate(endDate)
    }

    onTextFilterChanged = (e) => {
        this.props.setTextFilter(e.target.value)
    }

    onSortByChanged = (e) => {
        if (e.target.value == "amount") {
            this.props.sortByAmount()
        } else if (e.target.value == "date") {
            this.props.sortByDate()
        }
    }

    render() {
        return <div className="content-container">
            <div className="input-group">
                <div className="input-group__item">
                    <input
                        className="text-input"
                        type="text"
                        placeholder="Search expenses"
                        value={this.props.filters.text}
                        onChange={this.onTextFilterChanged} />
                </div>
                <div className="input-group__item">
                    <select
                        className="select"
                        value={this.props.filters.sortBy}
                        onChange={this.onSortByChanged}>
                        <option value="date">Date</option>
                        <option value="amount">Amount</option>
                    </select>
                </div>
                <div className="input-group__item">
                    <DateRangePicker
                        startDate={this.props.filters.startDate}
                        endDate={this.props.filters.endDate}
                        onDatesChange={this.onDatesChanged}
                        focusedInput={this.state.calendarFocused}
                        startDateId="startId1"
                        endDateId="endId1"
                        onFocusChange={this.onFocusChange}
                        numberOfMonths={2}
                        isOutsideRange={() => false}
                        showClearDates={true} />
                </div>
            </div>
        </div>
    }
}

const mapStateToProps = (state) => ({
    filters: state.filters
});

const mapDispatchToProps = (dispatch) => ({
    setStartDate: (startDate) => dispatch(setStartDate(startDate)),
    setEndDate: (endDate) => dispatch(setEndDate(endDate)),
    setTextFilter: (textFilter) => dispatch(setTextFilter(textFilter)),
    sortByAmount: () => dispatch(sortByAmount()),
    sortByDate: () => dispatch(sortByDate()),
})

export default connect(mapStateToProps, mapDispatchToProps)(ExpenseListFilters);