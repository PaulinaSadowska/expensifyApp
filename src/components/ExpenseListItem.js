import React from 'react';
import { connect } from 'react-redux';
import { removeExpense } from '../actions/expenses'
import {Link} from 'react-router-dom'

const ExpenseListItem = ({ description, amount, createdAt, id, dispatch }) => (
    <div>
        <Link to={`/edit/${id}`}>
            <h3>{description}</h3>
        </Link>
        <p>${(amount / 100).toFixed(2)} - {createdAt}</p>
        <button onClick={() => {
            dispatch(removeExpense({ id }))
        }}>Remove</button>
    </div>
);

export default connect()(ExpenseListItem)