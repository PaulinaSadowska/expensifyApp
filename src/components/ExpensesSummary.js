import React from 'react'
import { connect } from 'react-redux';
import selectExpenses from '../selectors/expenses';
import getExpensesTotal from '../selectors/expenses-total';
import numeral from 'numeral';
import expenses from '../selectors/expenses';

export const ExpensesSummary = (props) => {
    const { expenses } = props
    const totalPriceFormatted = numeral(getExpensesTotal(expenses)/100).format('$0,0.00')
    const expenseWord = (expenses.length===1) ? "expense" : "expenses"
    return <p>
        <h2>Viewing {expenses.length} {expenseWord} totalling {totalPriceFormatted}</h2>
    </p>  
};

const mapStateToProps = (state) => {
    return {
        expenses: selectExpenses(state.expenses, state.filters)
    }
}

export default connect(mapStateToProps)(ExpensesSummary);