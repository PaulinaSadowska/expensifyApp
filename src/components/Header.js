import React from 'react';
import { NavLink } from 'react-router-dom';

export const Header = () => (
    <header>
        <h1>Expensify</h1>
        <NavLink to="/" exact={true} activeClassName="is-active">Home page</NavLink>
        <NavLink to="/create" activeClassName="is-active">Create expense</NavLink>
    </header>
);