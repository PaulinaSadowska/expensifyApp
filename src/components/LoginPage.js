import React from 'react';
import { connect } from 'react-redux';

import { startLogIn } from '../actions/auth';

export class LoginPage extends React.Component {

    onSubmit = (e) => {
        e.preventDefault();
        this.props.startLogIn()
    }

    render() {
        return (
            <div>
                <h1>Log In</h1>
                <form onSubmit={this.onSubmit}>
                    <button>Log In</button>
                </form>
            </div>
        );
    }
}

const mapDispatchToProps = (dispatch) => ({
    startLogIn: () => dispatch(startLogIn())
})

export default connect(undefined, mapDispatchToProps)(LoginPage)