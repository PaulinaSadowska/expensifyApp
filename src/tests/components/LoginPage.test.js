import { LoginPage } from '../../components/LoginPage';
import React from 'react';
import { shallow } from 'enzyme';

test('should render login page', () => {
    const wrapper = shallow(<LoginPage startLogin={() => { }} />);
    expect(wrapper).toMatchSnapshot();
})

test('should logout button call logout axction', () => {
    const login = jest.fn()
    const wrapper = shallow(<LoginPage startLogin={login} />);
    wrapper.find('button').simulate('click')
    expect(login).toHaveBeenCalled()
})