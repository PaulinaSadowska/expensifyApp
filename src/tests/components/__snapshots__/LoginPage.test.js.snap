// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render login page 1`] = `
<div>
  <h1>
    Log In
  </h1>
  <form
    onSubmit={[Function]}
  >
    <button>
      Log In
    </button>
  </form>
</div>
`;
